src/platform/Tizen/DnssdImpl.cpp

```diff
@@ -55,6 +55,9 @@ constexpr const char * kEmptyAddressIpv6 = "0000:0000:0000:0000:0000:0000:0000:0
 // and the browsing callback is called multiple times (once for each service found).
 constexpr unsigned int kDnssdBrowseTimeoutMs = 250;
 
+// The number of miliseconds after which a mDNS resolve is considered failed.
+constexpr unsigned int kDnssdResolveTimeoutMs = 5000;
```

Comment:
Why?  The caller of this code will time out the resolve and try to cancel it after it decides it's taken too long, and the timeout there is 45s.  It used to be 15s, but that was too short in practice.

Cutting off resolves at 5s is mostly a guarantee that they will fail when they should not.

Reply:
@hyunuktak, can you respond to this?

Reply:
We can refer avahi's implementation. the timeout is defined in avahi's implementation [here](https://github.com/lathiat/avahi/blob/40f44806c5194e866ab574593891f3ea85a28e0c/avahi-core/resolve-address.c#L33) (5000ms). 
The trail from Matter SDK that leads to that timeout value in Avahi implementation is as follows:

1. [ChipDnssdResolve()](https://github.com/project-chip/connectedhomeip/blob/14ffac08583e5ee41c62270a2c4d58fea23eae1e/src/platform/Linux/DnssdImpl.cpp#L865)
2. [MdnsAvahi::Resolve()](https://github.com/project-chip/connectedhomeip/blob/14ffac08583e5ee41c62270a2c4d58fea23eae1e/src/platform/Linux/DnssdImpl.cpp#L665)
3. [avahi_service_resolver_new()](https://github.com/project-chip/connectedhomeip/blob/14ffac08583e5ee41c62270a2c4d58fea23eae1e/src/platform/Linux/DnssdImpl.cpp#L689)
4. [avahi_s_address_resolver_new()](https://github.com/lathiat/avahi/blob/40f44806c5194e866ab574593891f3ea85a28e0c/avahi-core/resolve-address.c#L278)
5. [avahi_s_address_resolver_prepare()](https://github.com/lathiat/avahi/blob/40f44806c5194e866ab574593891f3ea85a28e0c/avahi-core/resolve-address.c#L178)
6. [start_timeout()](https://github.com/lathiat/avahi/blob/40f44806c5194e866ab574593891f3ea85a28e0c/avahi-core/resolve-address.c#L246)

Also, note that the SDK has up to [3 retries](https://github.com/project-chip/connectedhomeip/blob/14ffac08583e5ee41c62270a2c4d58fea23eae1e/src/platform/Linux/DnssdImpl.cpp#L713) whenever resolution failure occurs, so you can expect the maximum timeout of 15s in 3 separate requests if they keep failing:
```
2023-04-28_00:56:12.280 | I | matter.sdk | [DL] Re-trying resolve
2023-04-28_00:56:17.304 | I | matter.sdk | [DL] Re-trying resolve
2023-04-28_00:56:22.337 | I | matter.sdk | [DL] Re-trying resolve
2023-04-28_00:56:27.369 | E | matter.sdk | [DL] Avahi resolve failed
```

@samadDotDev could you check this?

Reply:
The Linux impl does call into Avahi, and repeats 3 times, yes.  I don't know whether the Avahi 5s timeout is for the full resolve process (looking up SRV/TXT, doing the DNS lookup, etc), or just for parts of it.  I see the timer being restarted in some cases in the linked-to code (most notably if the initial unicast query fails and a multicast retry happens, per the RFC), so I expect it's longer than 5s in practice, but I would not guarantee that.

But I don't see the "3 retries" thing in the Tizen code.  Am I just missing it?

And more to the point, given that the higher level has its own timeout, why do you need one here?  I know the Linux impl has it, because it was created before the higher level was even written, but we know what the higher-level expectations are here now.... and should probably fix the Linux impl to not give up until the higher level code does either.

Reply:
He is going to check, so please wait a moment.
@samadDotDev Please check this.